modules/ssh/agent.go

```diff
@@ -0,0 +1,112 @@
+package ssh
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/crypto/ssh/agent"
+)
+
+type SshAgent struct {
+	stop       chan bool
+	stopped    chan bool
+	socketDir  string
+	socketFile string
+	agent      agent.Agent
+	ln         net.Listener
+}
+
+// Create SSH agent, start it in background and returns control back to the main thread
```

Comment:
You should tell the user that it's their responsibility to stop the agent, and how to do it.

Reply:
Added